File: Logic_HandsOn/C++/Problems on bit,File Handing,Pattern etc/program131.cpp
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Write a program which accept one number and position from user and check whether bit at that position is on or off. 
//		If bit is one return TURE otherwise return FALSE.
//
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////


#include<iostream>
using namespace std;

class Solution
{
	public:
		bool ChkBit(int iValue,int iPos)
		{
			int iMask = 0x00000001;
			int iResult = 0;

			iMask = iMask<<(iPos-1);
			iResult = iValue | iMask;

			if(iResult==iValue)
			{
				return true;
			}
			else
			{
				return false;
			}
		}
};

int main()
{
	int iValue=0,iPos=0;
	bool bRet=false;

	cout<<"Enter number: ";
	cin>>iValue;

	cout<<"Enter Position: ";
	cin>>iPos;

	Solution *sObj = new Solution();
	bRet = sObj->ChkBit(iValue,iPos);
	
	if(bRet == true)
	{
		cout<<"Bit is ON\n";
	}
	else
	{
		cout<<"Bit is OFF\n";
	}

	return 0;
}

/*
OUTPUT:

Enter number: 10
Enter Position: 2
Bit is ON

*/

File: Logic_HandsOn/Java/Problems on bit,file handling,String etc/Program131.java
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Write a program which accept one number and position from user and check whether bit at that position is on or off. 
//		If bit is one return TURE otherwise return FALSE.
//
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////


import java.util.Scanner;

class Solution
{
	public int iMask,iNo,iPos;
	public int iResult;

	Solution(int iNo,int iPos)
	{
		this.iNo = iNo;
		this.iMask = 0x00000001;
		this.iResult = 0;
		this.iPos = iPos;
	}

	boolean ChkBit()
	{

		iMask = iMask<<(iPos-1);
		iResult = iNo | iMask;

		if(iResult==iNo)
		{
			return true;
		}
		else
		{
			return false;
		}
	}	
}

class Program
{
	public static void main(String[] args)
	{
		boolean bRet=false;

		Scanner sObj = new Scanner(System.in);
		
		System.out.println("Enter number: ");
		int iValue = sObj.nextInt(); 

		System.out.println("Enter Position: ");
		int iPos = sObj.nextInt(); 
		
		Solution sObj1 = new Solution(iValue,iPos);
		bRet = sObj1.ChkBit();

		if(bRet==true)
		{
			System.out.println("Bit is ON");
		}
		else
		{
			System.out.println("Bit is OFF");
		}
	}	
}
/*
OUTPUT:

Enter number:
10
Enter Position:
2
Bit is ON

*/
















